zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -187,6 +196,10 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
     if (zkConnection == null) {
       throw new NullPointerException("Zookeeper connection is null!");
     }
+
+    String syncUpOnNewSession = System.getProperty(ZkSystemPropertyKeys.ZK_AUTOSYNC_ENABLED, ZK_AUTOSYNC_ENABLED_DEFAULT);
+    _syncOnNewSession = Boolean.parseBoolean(syncUpOnNewSession);
```

Comment:
Nit. This could be a constant: `static final boolean SYNC_ON_NEW_SESSION = Boolean.parseBoolean(System.getProperty(ZkSystemPropertyKeys.ZK_AUTOSYNC_ENABLED, ZK_AUTOSYNC_ENABLED_DEFAULT))`

Reply:
_syncOnNewSession is final not static as this allows per zkclient different _syncOnNewSession behavior. We may later support this from the ZkClient constructor. Let us just keep it this way.